incrementalbuild/src/main/java/io/takari/incrementalbuild/maven/internal/MavenBuildContextFinalizer.java

```diff
@@ -105,9 +106,24 @@ protected void failBuild(final Map<Object, Collection<Message>> messages)
         }
       }
     }
-    if (errorCount > 0) {
+
+    if (errorCount > 0 && shouldThrowException(contexts)) {
       throw new MojoExecutionException(errorCount + " error(s) encountered:\n" + errors.toString());
     }
+
+
+  }
+
+  private boolean shouldThrowException(List<AbstractBuildContext> contexts){
+    boolean foundFalse = false;
+    for (AbstractBuildContext context : contexts) {
+      if (context.getFailOnErrorState().equals(FailOnErrorState.TRUE)) {
```

Comment:
-1. One context should not decide how to deal with errors reported via other contexts.

It should be possible to change `MessageSinkAdaptor` above to only collect error messages from contexts with `failOnError=true`. I don't think `failBuild(..)` cares about any other messages. Note that `messageSink` should still get all messages.

Reply:
@ifedorenko But even if we have multiple context, the execution is unified. Sounds like the option that we should do here is throw an exception if contexts have different value. Is it ok?

Reply:
Yes, `IllegalStateException` in case of inconsistent `failOnError` is fine too.